### PR TITLE
feat(dlp): added a sample for deidentify table bucketing 

### PR DESCRIPTION
**Pull request by AarshD-crest**
## Description
- Added a sample for [deidentify table bucketing](https://cloud.google.com/dlp/docs/samples/dlp-deidentify-table-bucketing) with a unit test case.

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [X] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [X] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [X] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [X] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: dlp/snippets/deid/deid_table_bucketing.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// deIdentifyTableBucketing de-identifies data using table bucketing
+func deIdentifyTableBucketing(w io.Writer, projectID string, table *dlppb.Table) error {
+	// projectId := "your-project-id"
+	// table := "your-table-value"
+
+	//if table value is not passed, the default table will be used
+	if table == nil {
```
Comment (telpirion) [inline dlp/snippets/deid/deid_table_bucketing.go:32]: comment: remove this check and the `table` parameter. Given what you're demonstrating here, it's okay to construct the table in the sample.

File: dlp/snippets/deid/deid_test.go [hunk 2]
```diff
@@ -117,3 +119,54 @@ func TestDeidentifyExceptionList(t *testing.T) {
 	}
 
 }
+
+func TestDeidentifyTableBucketing(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	row1 := &dlppb.Table_Row{
```
Comment (telpirion) [inline dlp/snippets/deid/deid_test.go:126]: comment: Remove these tables from the test.

It's sufficient to embed the table in the sample, as it exists for demonstration purposes. That way customers know how to construct a table.

**Review by telpirion — CHANGES_REQUESTED** [review]

File: dlp/snippets/deid/deid_test.go [hunk 3]
```diff
@@ -117,3 +119,54 @@ func TestDeidentifyExceptionList(t *testing.T) {
...
+	}
+
+	buf := new(bytes.Buffer)
+	if err := deIdentifyTableBucketing(buf, tc.ProjectID, table); err != nil {
+		t.Errorf("deIdentifyTableBucketing: %v", err)
+	}
+	got := buf.String()
+	if want := "Table after de-identification"; !strings.Contains(got, want) {
+		t.Errorf("deIdentifyTableBucketing got %q, want %q", got, want)
+	}
```
Comment (annadata) [inline dlp/snippets/deid/deid_test.go:170]: Please see the Java example. And corresponding test dlp.snippets.DeIdentificationTests#testDeIdentifyTableBucketing in Java samples repo.
Please confirm that the bucketing is actually happening.

Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: Hii @annadata, 
Thank you for your suggestion and guidence, As you have suggested I have made the changes as per the java-code samples currently, just want to point out one thing for your reference that will not be aligned with existing go-code samples: 
- You can see that any go code sample has not any other return type except `error`. but for this sample's test case I have to compare with `expectedTable` , and for that, I have to provide return type `*dlppb.Table` alongside with error.

Comment (annadata) [reply dlp/snippets/deid/deid_test.go]: @AarshD-crest 
Sorry for not being clear. Please confirm that "75" in input is replaced with ""70:80".

File: dlp/snippets/deid/deid_table_bucketing.go (hunk 1 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_table_bucketing.go]: Done.

File: dlp/snippets/deid/deid_test.go (hunk 2 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: Done.

**Review by telpirion — APPROVED** [review]

File: dlp/snippets/deid/deid_test.go (hunk 3 above)
Comment (AarshD-crest) [reply dlp/snippets/deid/deid_test.go]: Hiii @annadata , I have updated the test case as per your suggestion. tell me if anything more is required.

**Review by annadata — APPROVED** [review]
LGTM

**Pull request merged**